### PR TITLE
update XSMM and bf8 benchmarking

### PR DESCRIPTION
**Pull request by alheinecke**

### CODE REVIEW

**Comment by alheinecke** [comment]
It seems we still need the 2nd libxsmm checkout as the pytorch extension needs to be build hat way.. perhaps we can somehow point this build script also to internals where cmake puts the checkout...

**Comment by rolfmorel** [comment]
The source gets placed in `python/build/cmake.linux-x86_64-cpython-3.9/_deps/cmake.linux-x86_64-cpython3.9/_deps/xsmm-src`. This path appears platform/python version dependent.

The .so gets placed in `python/triton/_C`.

That is probably enough information for the xsmm_py helper.

File: python/tutorials/03-matrix-multiplication-cpu.py [hunk 1]
```diff
@@ -441,6 +441,11 @@ def matmul(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor, M: int, N: int, K:
     'triton-cpu-single', 'triton-cpu', 'torch-cpu-native', 'torch-cpu-compile']
 LINE_NAMES = ['TritonCPU 1', 'TritonCPU', 'TorchCPU (native)', 'TorchCPU (compile)']
 LINE_STYLES = [('blue', '--'), ('blue', '-'), ('green', '--'), ('green', '-')]
+if DATA_TYPE == torch.float8_e5m2:
```
Comment (rolfmorel) [inline python/tutorials/03-matrix-multiplication-cpu.py:444]: Do we want to do this for BF8 regardless of target it runs on? I thought the FP8 benchmark did run on SPR with PyTorch.

**Review by rolfmorel — APPROVED** [review]
Except for 1 comment, LGTM.

File: python/tutorials/03-matrix-multiplication-cpu.py (hunk 1 above)
Comment (alheinecke) [reply python/tutorials/03-matrix-multiplication-cpu.py:444]: it runs at 1 GFLOPS, so the script would run in hours.  Did you see better numbers torch and torch.compile?

Comment (rolfmorel) [reply python/tutorials/03-matrix-multiplication-cpu.py:444]: I double checked this and indeed running with Torch is very slow. (And it wasn't a problem for me as I essentially had the same uncommited change locally as you have here.)

**Pull request merged**